### PR TITLE
[SYCL] Remove fusion scheduler integration

### PR DESCRIPTION
**Pull request by sommerlukas**
Remove the integration of kernel fusion from the SYCL runtime scheduler. This makes the implementation of the scheduler less complex and leaner.

As a consequence, the experimental kernel fusion feature is no longer supported, the extension proposal status has been updated.

Some stubs of the kernel fusion must remain for now to avoid ABI-break, removing those stubs in the next ABI-breaking window is tracked through #15184.

### CODE REVIEW

File: sycl/include/sycl/detail/property_helper.hpp
Comment (victor-eds) [inline sycl/include/sycl/detail/property_helper.hpp:36]: #15185?

File: sycl/include/sycl/ext/codeplay/experimental/fusion_wrapper.hpp
Comment (victor-eds) [inline sycl/include/sycl/ext/codeplay/experimental/fusion_wrapper.hpp:9]: Same

File: sycl/include/sycl/info/ext_codeplay_device_traits.def
Comment (victor-eds) [inline sycl/include/sycl/info/ext_codeplay_device_traits.def:5]: #15185

**Review by victor-eds — COMMENTED** [review]
NITs

File: sycl/include/sycl/info/ext_codeplay_device_traits.def
Comment (sommerlukas) [reply sycl/include/sycl/info/ext_codeplay_device_traits.def:5]: No, I've created an issue to track this in #15184.

File: sycl/include/sycl/ext/codeplay/experimental/fusion_wrapper.hpp
Comment (sommerlukas) [reply sycl/include/sycl/ext/codeplay/experimental/fusion_wrapper.hpp:9]: No, I've created an issue to track this in https://github.com/intel/llvm/issues/15184.

File: sycl/include/sycl/detail/property_helper.hpp
Comment (sommerlukas) [reply sycl/include/sycl/detail/property_helper.hpp:36]: No, I've created an issue to track this in https://github.com/intel/llvm/issues/15184.

Comment (victor-eds) [reply sycl/include/sycl/detail/property_helper.hpp:36]: Perfect!

File: sycl/doc/extensions/deprecated/sycl_ext_codeplay_kernel_fusion.asciidoc
Comment (EwanC) [inline sycl/doc/extensions/deprecated/sycl_ext_codeplay_kernel_fusion.asciidoc:1]: Might need to update the SYCL-Graph spec link to kernel fusion - https://github.com/intel/llvm/blob/db01e68b7575f2d8002ecf6d09e273c8ec207b40/sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc#7127-sycl_ext_codeplay_kernel_fusion spec

File: sycl/source/detail/scheduler/scheduler.cpp
Comment (jchlanda) [inline sycl/source/detail/scheduler/scheduler.cpp:9]: NIT, maybe at the same time you could align the system vs local include, eve though it's not touched in this diff?

**Review by jchlanda — APPROVED** [review]
👍

File: sycl/doc/extensions/deprecated/sycl_ext_codeplay_kernel_fusion.asciidoc
Comment (EwanC) [inline sycl/doc/extensions/deprecated/sycl_ext_codeplay_kernel_fusion.asciidoc:1]: Why is this extension moving to the "depreciated" extensions folder rather than "removed"? In my mind, if it was depreciated it would still be available, but not advised to to be used.

**Review by EwanC — APPROVED** [review]

**Review by sergey-semenov — APPROVED** [review]

**Comment by KseniyaTikhomirova** [comment]
@sommerlukas could you please resolve conflicts with sycl branch? it blocks merge.

**Comment by sommerlukas** [comment]
> @sommerlukas could you please resolve conflicts with sycl branch? it blocks merge.

@KseniyaTikhomirova I merged and resolved conflicts. However, the merge is still blocked because of missing review from @intel/dpcpp-specification-reviewers. 

@intel/dpcpp-specification-reviewers Friendly ping. This PR moves an experimental extension to "removed" and updates a link to that extension in another document.

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc
Comment (gmlueck) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:1930]: Should this section be rewritten?  It doesn't really make sense to explain the interaction with an extension that doesn't exist.

Comment (sommerlukas) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc:1930]: We could probably remove this entire section. What do you think @EwanC?

Comment (sommerlukas) [reply sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]: Checked with @EwanC and decided to remove this section. Done in the latest commit.

**Review by gmlueck — APPROVED** [review]
spec changes LGTM

**Comment by sommerlukas** [comment]
> @sommerlukas - There seems to be some build issues in post-commit, seemingly due to unused arguments. Could you please have a look ASAP?

@steffenlarsen  Should be fixed in https://github.com/intel/llvm/pull/15284.

**Pull request merged**
